### DISCUSSION TITLE
Energy Dashboard Pellet Consumption Integration

### DISCUSSION THREAD

**Discussion by lucaferrucci** [dominikamann/oekofen-pellematic-compact#36] [Q&A] [answered]
Hi Dominque, how are you?
Were you able to benefit from the new pellematic settings?
Now okofen has built an obviously very high quality heat pump.
I managed to install the photovoltaic system at home, therefore taking a good step forward.
I'm messing around with the energy dashboard in HomeAssistant to implement everything and an idea came to me.
There is the gas consumption panel, which is of little use in this era, it would be nice to be able to transform it into the consumption of pellets both in Kg and Kw of thermal power, since having the possibility of finding the value of daily consumption would really be nice, then for solar thermal I also have a counter for the kW generated, so having the flow of the sun's power and the pellets consumed, like for electricity, would be really fantastic.
what do you think about it? doable?

**Comment by dominikamann** [comment, marked as answer]
Hi @lucaferrucci, 

Sounds good. Thank you, I am doing well. I hope you too?
"Wood" and "pellet" are currently not supported by HomeAssisant but there is a workarround:  https://community.home-assistant.io/t/energy-adding-pellets-and-wood-to-gaz-and-electricity/335031. I think you can use this to achieve your goal!

**A easy way is to go to:**

Settings -> Devices -> Helpers -> Add a new template sensor / helper

Template: `{{(states('sensor.pelletheizung_pellet_1_usage_today') | float)}}` / Total Increasing / kwH / Energy

![grafik](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/afe9b968-d9a2-49db-b58a-8a0ea7c06e33)

After that you are able to configure this Helper-Entity as "Gas usage" in the Energy-Dashboard. Now you have your consumption visible. This should work. 

![grafik](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/e7cced7b-d60f-4e18-9da3-b8a6adad3b13)


Let me know if it is working for you!

**Reply by dominikamann** [reply]
@lucaferrucci could you try it out?

**Reply by dominikamann** [reply]
Now it was cold enough and my pellet heater started. It works as expected. You can see the pellet consumption in the default energy dashboard panel.

![grafik](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/fd274333-5981-4964-8380-da5468df9880)

**Comment by lucaferrucci** [comment]
Hi Dominique,
 I tried, but strangely as soon as I activated I had problems with the other consumer entities, I have to try again, maybe I did something wrong

**Comment by lucaferrucci** [comment]
Hi Dominiq, I configured the sensor, but obviously now it's 27 degrees here and it's very hot, so since the pellematic is off the value is zero, we'll see what will happen when it's operational, I'm trying to understand how to also enter the performance of the solar thermal

**Comment by lucaferrucci** [comment]
Hi Dominik, I'm trying to create an energy panel similar to the standard homeassistant dashboard but with the data coming from the compact, I'm creating the formula to quantify the energy in kW of the puffer and therefore consider it as a battery, the formula is determined by the temperature of the puffer in the TPO probe, the external temperature and the capacity in liters of the puffer.
In your integration, the performance data of the solar system is not taken via json, there is an accountant that calculates the thermal power generated by the panels, both in real time and the daily and annual performance.
There should be all the data to consider our heating system as a photovoltaic one.
I attach the screenshots, if you need I can also send you the json file to identify the solar thermal fields.
you calculate the energy in kW in the puffer as per my formula, considering the ambient temperature of the house and the temperature values


sensor:
  - platform: template
    sensors:
      energia_termica_puffer:
        friendly_name: "Energia Termica nel Puffer"
        unit_of_measurement: "kWh"
        value_template: "{{ (states('sensor.pellematic_buffer_storage_temperature_upper_area') | float - states('sensor.ntsalone_temperature') | float) * 2.3 }}"    
    
![Screenshot 2023-10-04 alle 09 41 53](https://github.com/dominikamann/oekofen-pellematic-compact/assets/110548623/fc2eb792-72ab-45a1-8f37-cfdb09250d9f)
![Screenshot 2023-10-04 alle 09 27 02](https://github.com/dominikamann/oekofen-pellematic-compact/assets/110548623/ead80e39-9a83-4605-9369-8946d0fe792b)

[all.pdf](https://github.com/dominikamann/oekofen-pellematic-compact/files/12801887/all.pdf)

**Reply by dominikamann** [reply]
@lucaferrucci so the following fields are missing? 

![272532770-fc2eb792-72ab-45a1-8f37-cfdb09250d9f](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/50cb26f8-9b71-4da9-95e7-7ed17837419a)

Can you please send me the JSON group and names?

**Comment by lucaferrucci** [comment]
Hi Dominik, how are you? Yes, I also activated the consumption of pellets, but as you can imagine in southern Italy, luckily we still have 24 degrees celsius, only at night to cool the puffer I discharge some heat in the house, but the compect is safely stored away

**Comment by lucaferrucci** [comment]
Yes Dominik
￼

**Reply by dominikamann** [reply]
I'm fine thanks. I hope you too? I can not open the attachment. Are you replying by email? Can you try directly on https://github.com/dominikamann/oekofen-pellematic-compact/discussions/36?

**Comment by lucaferrucci** [comment]
ok

**Reply by lucaferrucci** [reply]
[jsonForDominik.webarchive.zip](https://github.com/dominikamann/oekofen-pellematic-compact/files/12878659/jsonForDominik.webarchive.zip)

**Comment by lucaferrucci** [comment]
[last.txt](https://github.com/dominikamann/oekofen-pellematic-compact/files/12878769/last.txt)

**Reply by dominikamann** [reply]
Thanks, as soon as I have some time I will take a look and implement the missing sensors.

**Reply by dominikamann** [reply]
@lucaferrucci i checked the JSON but iam not able to find the values. For me it seems that the JSON-Inteface of Ökofen does not provide it :/

![grafik](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/dad9b06c-4cb6-4bb1-bd58-b44519f4fab8)

**Comment by lucaferrucci** [comment]
Hi Dominik 

 hi Dominik these should be, they refer to l_day to the "today" return and yesterday to the "yesterday" return, you see "0" on l_day because it wasn't a day without sun

**Reply by dominikamann** [reply]
Hi  @lucaferrucci,

I double checked the code. "L_day" and "L_yesterday" is already implement as "Solar Gain Today" and "Solar Gain Yesterday". So it should be already available for you. Did you saw this Sensors?

**Comment by lucaferrucci** [comment]
![Screenshot 2023-10-19 alle 09 41 23](https://github.com/dominikamann/oekofen-pellematic-compact/assets/110548623/34024d5d-6093-4858-b8c0-7ac0a3ae8d69)

**Comment by lucaferrucci** [comment]
It’s now
￼

**Comment by lucaferrucci** [comment]
Hi Domink, yes I have seen them, but they are not available on the homeassistant energy panel, I don't know why

**Reply by dominikamann** [reply]
Hi @lucaferrucci,

you can do the same as I suggested above for the Pellet-Usage. You only need to create a new template sensor for the "Solar Gain Today" Sensor. Let me know if it works!

**Sample:**

Settings -> Devices -> Helpers -> Add a new template sensor / helper

Template: {{(states('sensor.pelletheizung_pellet_1_solar_gain_today') | float)}} / Total Increasing / kwH / Energy

(You need to adjust the sensor name)

![image](https://github.com/dominikamann/oekofen-pellematic-compact/assets/29973737/f2204816-8eb9-4557-9702-48b0c1d5ff97)

**Comment by lucaferrucci** [comment]
Sensor None has device class 'energy', state class 'total_increasing' unit 'kWh' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: 'Template: 0.0 / Total Increasing / kwH / Energy' (<class 'str'>)

**Comment by lucaferrucci** [comment]
for pellet is same, value is already "0"

**Comment by dominikamann** [comment]
FYI: The above solution is now also working for lucaferrucci. We have set it up together.
